app/routes/user/endorsements.server.ts

```diff
@@ -0,0 +1,214 @@
+/*!
+ * Copyright © 2022 United States Government as represented by the Administrator
+ * of the National Aeronautics and Space Administration. No copyright is claimed
+ * in the United States under Title 17, U.S. Code. All Other Rights Reserved.
+ *
+ * SPDX-License-Identifier: NASA-1.3
+ */
+
+import { tables } from '@architect/functions'
+import { getSubmitterUsersFromCognito } from '~/lib/utils'
+import { getUser } from '../__auth/user.server'
+
+// models
+export type EndorsementRequest = {
+  endorserEmail: string
+  endorserSub: string
+  requestorEmail: string
+  requestorSub: string
+  created: number
+  status: EndorsementStates
+}
+
+export type EndorsementStates = 'approved' | 'pending' | 'rejected' | 'reported'
+
+export class EndorsementsServer {
+  #sub: string
+
+  private constructor(sub: string) {
+    this.#sub = sub
+  }
+
+  static async create(request: Request) {
+    const user = await getUser(request)
+    if (!user)
+      throw new Response('not signed in', {
+        statusText: 'not signed in',
+        status: 403,
+      })
+    return new this(user.sub)
+  }
+
+  async createEndorsementRequest(request: EndorsementRequest) {
+    // Checking endorser sub:
+    if (!this.requestedEndorserIsValid(request.endorserSub))
+      throw new Response('Requested Endorser is not a verified submitter', {
+        statusText: 'Bad Request',
+        status: 400,
+      })
+    const db = await tables()
```

Comment:
We need to check that if an endorsement request with matching subs already exists, then it is in the pending state. That way, the user cannot overwrite a request that was already made, unless doing so would be idempotent.